server/api_test.go

```diff
@@ -0,0 +1,110 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/mattermost/mattermost-server/plugin"
+	"github.com/mattermost/mattermost-server/plugin/plugintest"
+	"github.com/mattermost/mattermost-server/plugin/plugintest/mock"
+	"github.com/stretchr/testify/assert"
+)
+
+var (
+	basicConfig = &configuration{
+		OAuthClientID:       "client-id",
+		OAuthClientSecret:   "client-secret",
+		WebexAuthorizeURL:   "https://api.ciscospark.com/v1/authorize",
+		WebexAccessTokenURL: "https://api.ciscospark.com/v1/access_token",
+		EncryptionKey:       "KoTkgdxmgP2XnJHRKsf_Dce-IZwP3nuX",
+	}
+
+	baseConnectedRequest = httptest.NewRequest("GET", "/connected", nil)
+)
+
+func TestConnected_Fail_NotAuthorizedNoHeaderMMUserID(t *testing.T) {
```

Comment:
We normally use only CamelCasing without any snake casing in the tests names. And probably here would make sense something like:
func TestConnected(t *testing.T) {
    ...
    t.Run("Fail with not authorized no header MMUserID", func(t *testing.T) {
        ...
    })
    t.Run("Fail with Webex Not Connected", func(t *testing.T) {
        ...
    })
    t.Run("Success with Webex Connected", func(t *testing.T) {
        ...
    })
}
```go

```